CHANGELOG.md

```diff
@@ -1,5 +1,92 @@
 # Changelog
 
+## Pelion Edge 2.2 - January 2021
+
+The primary features in this release:
+
+* [edge-core] Updated Edge Core to [0.15.0](https://github.com/PelionIoT/mbed-edge/blob/master/CHANGELOG.md#release-0150-2021-1-12).
+  * Moved the mbed-edge-core recipe to its own meta layer - [meta-mbed-edge](https://github.com/PelionIoT/meta-mbed-edge).
+  * The new FOTA update framework library is supported on platforms `imx8mmevk` and `uz3eg-iocc` but not on `raspberrypi3`. To compile with this library, add the `MBED_EDGE_CORE_CONFIG_FIRMWARE_UPDATE="ON"`, `MBED_EDGE_CORE_CONFIG_FOTA_ENABLE="ON"` and `MBED_EDGE_CORE_CONFIG_CURL_DYNAMIC_LINK="ON"` Bitbake parameters to local.conf.
```

Comment:
This needs to be sync'd with @esajaa.

Afte running multiple tests, I've found that LmP sources the environment and variables in a different way than Yocto. I also think that the job that @OPpuolitaival and @esajaa have been working on add the variables in a different location.

Reply:
What's the difference? Both yocto and lmp builds add the variables to the local.conf? Or so I understood from Yocto quick start  https://developer.pelion.com/docs/device-management-edge/2.1/quick-start/yocto-quick-start.html

Reply:
@esajaa Unless something's changed within the last day, the way LmP sources its variables is a bit more involved than Yocto, and a bit strange.

Should it be mentioned that the location for LmP's 'local.conf' resides within a hidden directory?

I've also had mixed experiences with setting your own variables within `.repo/manifests/conf/local.conf`, mainly because it seems that when LmP sources `setup-environment` for the first time, it's doing a lot more than what a Yocto env would do.

I've been testing it today, but have had mixed results, but definitely know that a local.conf file is not available prior to invoking the `source setup-environemnt build-factory` command.

Reply:
Ah, I think I see where the confusion comes from. In theory there are now two local.conf.samples. One comes from meta-pelion-edge, and one comes from repo containing lmp-manifest (since lmp uses the local.conf a bit different way). LmP doesn't utilize the local.conf.sample from meta-pelion-edge, so any changes there won't be applied. Possible new variables have to be added to this .repo folder local.conf.sample.

local.conf file should not be present before running setup-environment in any type build. What setup-environment does is that it pulls some local.conf.sample, and places it to the build folder with name local.conf. It may be that lmp setup-environment does some extra things with variables, but that's at least the behavior with local.conf

After the local.conf is set, running the setup-environment doesn't pull the local.conf.sample anymore

Reply:
> local.conf file should not be present before running setup-environment in any type build. What setup-environment does is that it pulls some local.conf.sample, and places it to the build folder with name local.conf

Yup. It's this nuanced detail that I think should be present in some way. Otherwise anyone thinking that they can set things using the normal local.conf method will have issues.

Reply:
Resolved with b016b00867efe85d73d6deb1808f89e48b2ec337